app/components/channel_intro/index.js

```diff
@@ -24,7 +24,7 @@ function makeMapStateToProps() {
                 return '';
             }
 
-            return channel.name.split('__').find((m) => m !== currentUserId);
+            return channel.name.split('__').find((m) => m !== currentUserId) || currentUserId;
```

Comment:
Why are you assuming that the teammate id here is currentUserId ?

Reply:
For own DM channel's name, it is like `8xauxmth4tnb3jrky4thfkspay__8xauxmth4tnb3jrky4thfkspay` where `8xauxmth4tnb3jrky4thfkspay` is self's user ID. It will return no match so the currentUserId is the the teammate id.
I think it's better to rely on `channel.teammate_id` when channel type is DM, then get the user profile out of it, than to have this function `getChannelMembersForDm` and `getOtherUserIdForDm` like what I did on first commit.

Reply:
actually we cannot rely on `channel.teammate_id` as we should be removing that from the getChannel selector as is modifying the object and creating a new object every time is being called so the selector is actually not memoizing anything.

That said, I think this is ok